controllers/release/adapter.go

```diff
@@ -802,19 +802,67 @@ func (a *adapter) cleanupProcessingResources(pipelineRun *tektonv1.PipelineRun,
 	}
 
 	if pipelineRun != nil {
-		if controllerutil.ContainsFinalizer(pipelineRun, metadata.ReleaseFinalizer) {
-			patch := client.MergeFrom(pipelineRun.DeepCopy())
-			removedFinalizer := controllerutil.RemoveFinalizer(pipelineRun, metadata.ReleaseFinalizer)
-			if !removedFinalizer {
-				return fmt.Errorf("finalizer not removed")
-			}
-			err := a.client.Patch(a.ctx, pipelineRun, patch)
-			if err != nil {
-				return err
-			}
+		if err := a.safelyRemovePipelineRunFinalizer(pipelineRun); err != nil {
+			return fmt.Errorf("failed to remove finalizer from PipelineRun %s/%s: %w",
+				pipelineRun.Namespace, pipelineRun.Name, err)
+		}
+	}
+
+	return nil
+}
+
+// safelyRemovePipelineRunFinalizer removes the release finalizer efficiently.
```

Comment:
I’m not sure about the method name and the description here. The terms “safely” and “efficiently” feel a bit vague, it’s not obvious what they mean in terms of behavior. I think we could simplify by just naming it `removePipelineRunFinalizer`, what do you think?